### DISCUSSION TITLE
Duplication over providers array

### DISCUSSION THREAD

**Discussion by mjbriggs** [CMSgov/price-transparency-guide#278] [Q&A] [answered]
Can the same tin appear more than once within a single providers array?

For example, I am referring to a scenario like so:
```     
     "provider_groups": [{
        "npi": [1111111111, 2222222222, 3333333333, 4444444444, 5555555555],
        "tin":{
          "type": "ein",
          "value": "11-1111111"
        }
      },{
        "npi": [1111111111, 6666666666],
         "tin":{
          "type": "ein",
          "value": "11-1111111"
        }
      }],
```

**Comment by shaselton-usds** [comment, marked as answer]
I'm trying to better understand this. Taking your example:
```json
     "provider_groups": [{
        "npi": [1111111111, 2222222222, 3333333333, 4444444444, 5555555555],
        "tin":{
          "type": "ein",
          "value": "11-1111111"
        }
      },{
        "npi": [1111111111, 6666666666],
         "tin":{
          "type": "ein",
          "value": "11-1111111"
        }
      }],
```
Is there any reason why it wouldn't be?:
```json
     "provider_groups": [{
        "npi": [1111111111, 2222222222, 3333333333, 4444444444, 5555555555, 6666666666],
        "tin":{
          "type": "ein",
          "value": "11-1111111"
        }
      }],
```

But to directly answer your question, "no" would be the answer -- it just would be inefficient in terms of space.

**Reply by mjbriggs** [reply]
Thank you for the response. 

We are trying to balance runtime performance vs file size. Given the technologies we are using, it could be meaningfully faster to print data that could be coalesced together, rather than actually coalescing and then printing. 

As I am thinking about it, I have a broader question about data duplication within a negotiated rate file. 

Within a health plan, there can be multiple prices for a billing code. These prices come from contracted rates. Contracted rates will live inside a contract between a healthcare organization and a payer. There can be multiple prices for a billing code within a health plan, therefore, we can assume that there could be multiple contracts represented within a health plan. Additionally, within a health plan, there can be multiple sources of prices for a single billing code. Being able to price and print per contract would yield tremendous scalability improvements. 

Obviously, at some point, these smaller pricing units (either data or the associated files), need to be put into one file under the umbrella of a single health plan. Allowance for partially duplicative data (like in the example above) provides a lot more flexibility on the implementation side of things as we try to balance file size, data generation speed, and scalability.
